### PR TITLE
chore: add contract deployment script and configs

### LINKED ISSUE #285: [document]A complete forcerelay operation document
A complete forcerelay operation document, can be found in Hermes (IBC CLI) document, supplemented related documentation, related documentation Link: https://hermes.informal.systems/documentation/commands/path-setup/clients.html

### PR DESCRIPTION
**Pull request by ashuralyk**
Closes: #285 

## Description

change logs:
1. make deployment process automatically through `ckb-cli`, for the utility of manual deployment
2. move the creation of `connections-cell` from the `ibc-test` to relayer core logic
3. fix a bug which is a wrong use of query_channels in CKB endpoint

### CODE REVIEW

**Comment by ashuralyk** [comment]
I'm writing them right in the Forcerelay repo, if it's more suitable to move them into the contract repo?

**Comment by Flouse** [comment]
> I'm writing them right in the Forcerelay repo, if it's more suitable to move them into the contract repo?

I think it's okey.
It's appropriate to organize the integration-test deployment guild in the Forcerelay repo.

**Comment by jjyr** [comment]
image

Please config your editor to add a blank line at the end of file.

https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log

**Comment by ashuralyk** [comment]
> Please config your editor to add a blank line at the end of file.

@jjyr it's bee confirmed

**Comment by ashuralyk** [comment]
@jjyr https://github.com/synapseweb3/forcerelay/actions/runs/6478088104/job/17601648711?pr=345#step:16:418 do you know anything about this error ?

**Comment by blckngm** [comment]
> @jjyr https://github.com/synapseweb3/forcerelay/actions/runs/6478088104/job/17601648711?pr=345#step:16:418 do you know anything about this error ?

This is probably because the sudt transfer packet is not relayed to axon.

**Comment by ashuralyk** [comment]
> > @jjyr https://github.com/synapseweb3/forcerelay/actions/runs/6478088104/job/17601648711?pr=345#step:16:418 do you know anything about this error ?
> 
> This is probably because the sudt transfer packet is not relayed to axon.

can this be fixed?

File: tools/ibc-test/contracts/deployment/upload.sh
Comment (Flouse) [inline tools/ibc-test/contracts/deployment/upload.sh:1]: file name suggestion:
upload -> deploy

Comment (Flouse) [reply tools/ibc-test/contracts/deployment/upload.sh:1]: ### suggestion
Should we specify the version (tag or commit) of forcerelay-ckb-contracts in the deployment?
I think it's more clearer to know which contract version you are deploying and testing.

Comment (ashuralyk) [reply tools/ibc-test/contracts/deployment/upload.sh:1]: indeed, and the version file is under the contracts directory outside

**Review by blckngm — APPROVED** [review]

**Pull request merged**